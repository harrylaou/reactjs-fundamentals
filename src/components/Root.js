import React from 'react'
import { Router, hashHistory } from 'react-router'
import configureRoutes from '../config/Routes'
import { Provider } from 'react-redux'
import '../../public/css/main.css'

const Root = ({ store }) => {
  const Routes =  configureRoutes(store)
  return (
    <Provider store={store}>
      <Router history={ hashHistory }>
        {Routes}
      </Router>
    </Provider>
  )
}

export default Root
