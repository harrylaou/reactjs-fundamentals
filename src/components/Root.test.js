import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import Root from './Root'

import FakeConfigureStore from 'redux-mock-store'

describe('<Root />', () => {
  it('should render with default props', () => {

    const mockStore = FakeConfigureStore()
    const store = mockStore()

    const wrapper = shallow(<Root store={store}/>)

    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})