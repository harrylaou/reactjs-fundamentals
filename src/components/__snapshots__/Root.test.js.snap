// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Root /> should render with default props 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Router
    history={
      Object {
        "__v2_compatible__": true,
        "createHref": [Function],
        "createKey": [Function],
        "createLocation": [Function],
        "createPath": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "listen": [Function],
        "listenBefore": [Function],
        "push": [Function],
        "pushState": [Function],
        "registerTransitionHook": [Function],
        "replace": [Function],
        "replaceState": [Function],
        "setState": [Function],
        "transitionTo": [Function],
        "unregisterTransitionHook": [Function],
      }
    }
    render={[Function]}
  >
    <Route>
      <Route
        component={[Function]}
        onEnter={[Function]}
        path="/"
      >
        <Route
          component={[Function]}
          path="users"
        >
          <Route
            component={[Function]}
            path=":username"
          />
        </Route>
        <Route
          component={[Function]}
          path="workshops"
        >
          <Route
            component={[Function]}
            path=":workshop"
          />
        </Route>
        <IndexRoute
          component={[Function]}
        />
      </Route>
      <Route
        component={[Function]}
        path="login"
      />
    </Route>
  </Router>
</Provider>
`;
